runtime/src/bank/builtins/mod.rs

```diff
@@ -1,110 +1,78 @@
-use {
-    solana_program_runtime::invoke_context::BuiltinFunctionWithContext,
-    solana_sdk::{
-        bpf_loader, bpf_loader_deprecated, bpf_loader_upgradeable, feature_set, pubkey::Pubkey,
-    },
-};
-
-/// Transitions of built-in programs at epoch bondaries when features are activated.
-pub struct BuiltinPrototype {
-    pub feature_id: Option<Pubkey>,
-    pub program_id: Pubkey,
-    pub name: &'static str,
-    pub entrypoint: BuiltinFunctionWithContext,
-}
+pub mod prototypes;
 
-impl std::fmt::Debug for BuiltinPrototype {
-    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        let mut builder = f.debug_struct("BuiltinPrototype");
-        builder.field("program_id", &self.program_id);
-        builder.field("name", &self.name);
-        builder.field("feature_id", &self.feature_id);
-        builder.finish()
-    }
-}
-
-#[cfg(RUSTC_WITH_SPECIALIZATION)]
-impl solana_frozen_abi::abi_example::AbiExample for BuiltinPrototype {
-    fn example() -> Self {
-        // BuiltinPrototype isn't serializable by definition.
-        solana_program_runtime::declare_process_instruction!(MockBuiltin, 0, |_invoke_context| {
-            // Do nothing
-            Ok(())
-        });
-        Self {
-            feature_id: None,
-            program_id: Pubkey::default(),
-            name: "",
-            entrypoint: MockBuiltin::vm,
-        }
-    }
-}
+pub use prototypes::{BuiltinPrototype, StatelessBuiltinPrototype};
```

Comment:
I'm a bit ambivalent about this re-export. What is your reasoning for it?

Reply:
Merely to preserve existing imports on other files, nothing more. I can cut this and use `runtime::builtins::prototypes::_` in those areas. I'm not sure how many files that changes, but should only be a few across bank, snapshots, and ledger.